### DISCUSSION TITLE
dict type hints should only take 1 or 2 inputs

### DISCUSSION THREAD

**Discussion by randolf-scholz** [python/typing#1247] [General]
It is weird that it is allowed to write something like `dict[str, int, float]` without raising an error.

A suggestion:

- `dict` ⇝ `dict[Any, Any]`
- `dict[key_type]` ⇝ `dict[key_type, Any]`
- `dict[key_type, value_type]`
- `dict[...]` with ≥3 arguments raises Error.

**Comment by erictraut** [comment]
Type checkers like mypy or pyright will generate an error if you supply too many type arguments. The Python runtime doesn't enforce this because generic types are erased at runtime.
